### ISSUE TITLE
Move current theme view files to better location

### ISSUE THREAD

**Issue by jayroh** [jayroh/thredded#123]
Currently "theme" files live in the dummy app under `spec/dummy/app/themes/thredded/<theme name>`. Each theme directory contains both the views and the assets. So for a "default theme" that would be

```
spec/dummy/app/themes/thredded/default/views/
spec/dummy/app/themes/thredded/default/assets/images/
spec/dummy/app/themes/thredded/default/assets/stylesheets/
spec/dummy/app/themes/thredded/default/assets/javascripts/
```

I will admit that I *do* like that they are all contained together in one bundle. To me, that's nice.
I do not like that they are in `spec/dummy/app/themes`, though.

Where should these go, instead? And is it not worth it to bundle them together, and instead default to the usual rails conventions (for ONE set of views and assets) of:

```
app/views/<engine name>/
app/assets/stylesheets/<engine name>.scss
app/assets/stylesheets/<engine name>/*
app/assets/javascripts/<engine name>.js
app/assets/javascripts/<engine name>/*
```

If I were to provide more than one "theme" (set of views and assets), though - how should those be packaged?

```
app/theme/<theme name>/views/<engine name>/
app/theme/<theme name>/assets/stylesheets/<engine name>.scss
app/theme/<theme name>/assets/stylesheets/<engine name>/*
app/theme/<theme name>/assets/javascripts/<engine name>.js
app/theme/<theme name>/assets/javascripts/<engine name>/*
```

?

@glebm @saturnflyer @TheDudeWithTheThing - would love your input!

**Comment by glebm** [comment]
The standard Rails location is best because it allows for overriding only the files one would want to modify, not copying everything as it is now. 

> I will admit that I do like that they are all contained together in one bundle. 

If a them is in a separate gem, it would be in one bundle :) 

E.g. If a gem is named `thredded-theme-awesome`, the location would be 

```ruby
app/views/thredded/theme/awesome/
etc
```

**Comment by jayroh** [comment]
@glebm is the voice of reason!

**Comment by TheDudeWithTheThing** [comment]
he's got me beat!

**Comment by jayroh** [comment]
After thinking about this a bunch the correct decision, per @glebm's suggestion, is to adhere to rails' conventions and go from there. There are some considerations though. I do not want to *require* anyone using this gem to use a particular sass/js pre-processor.

If someone were to install the gem and prefer less and coffeescript, I wouldn't want to force them to use sass, bourbon, and babeljs (sprockets-es6). Therefore the default should be to work with only a bare(-ish) subset of the view files -- no references to sass/css or js/eS6 assets.

One step above that would be to install the gem via the generator with "themed" support -- that would include references to the sass and js assets.

An extra option could be to *copy* the views and/or the assets to the parent app. How would this look? Take a look and tell me if this makes sense.

```
rails g thredded:install  # only the initializer copied over.
rails g thredded:install --themed # initializer + sass/es6 dependencies + layout with references to assets
rails g thredded:install --copy-theme # initializer + sass/es6 depencies + all views, layout, assets
```

Is this too granular? Am I missing anything?

**Comment by jayroh** [comment]
@TheDudeWithTheThing @glebm @saturnflyer 

would love y'all's opinion on that :point_up:. Jim especially considering you just tried to integrate it and found a bunch of holes in the assumptions.

**Comment by TheDudeWithTheThing** [comment]
That seems ok but this is not an area I know much about. 

From an any project perspective I would expect that something with theme support would build me out or include a skeleton to start from so it sounds like your copy-theme option would take care of that.

I'd have to try it to really see if it "feels right"

**Comment by glebm** [comment]
All asset dependencies (sass/es6) can simply be the gem's dependencies. While it might mean bloating the main app's dependencies slightly, this can be resolved by providing the theme as a separate gem. I don't think the bloat is large enough to be warrant workarounds like this. This would also make the `--themed` option redundant, as there is the layout would be served directly from the gem by default.

**Comment by jayroh** [comment]
Yeah I see your point. It's not something that couldn't be addressed in the future if it was ever a concern.

"MVP" and all, right?

**Comment by glebm** [comment]
Also an issue of developer time being a much larger cost than RAM, network, and storage of 500KB :+1:

**Comment by jayroh** [comment]
I think PR https://github.com/jayroh/thredded/pull/127 should address everything we've discussed here.

Feedback welcome and appreciated!

**Comment by jayroh** [comment]
merged with #127
